person.rb

```diff
@@ -15,9 +18,13 @@ def can_use_services?
     false
   end
 
+  def validate_name
+    @name = @corrector.correct_name(@name)
```

Comment:
It is not needed except on `@corrector` unless you add an `attr_reader :corrector`.

I have a little pattern that I follow. I'm not particularly eager to use `@`, so I create the reader or accessors.

But instead of making them all public accessors (I sometimes do not want to allow code outside the class to change or read the state of these instance variables), I will make them private accessors.

So this means I will use the `private` keyword and then place the accessor there.

```ruby
  # after all (normal) public methods are declared, I use the private keyword to declare private methods:
  private

  attr_reader :corrector
```

This makes it much easier to refactor the code later on, and it looks nicer to me.

Reply:
thanks, i was a bit confused about this. i thought it was calling the initialize parameter value without the @